common/kvstore/src/main/java/org/apache/spark/util/kvstore/RocksDB.java

```diff
@@ -418,6 +414,33 @@ private byte[] getTypeAlias(Class<?> klass) throws Exception {
     return alias;
   }
 
+  private void stopCleanupThread() throws InterruptedException {
+    if (cleaningThread != null) {
+      stopped = true;
+      cleaningThread.join();
+    }
+  }
+
+  private void keepCleaning() {
+    boolean lastIsPresent = false;
```

Comment:
Added `lastIsPresent` to ensure all `RocksDBIterator`s dropped by GC are closed  , otherwise `testAllRocksDBIteratorNotExplicitClosed` will lead JVM crash